### ISSUE TITLE
read_reaper_est_file: IndexError: too many indices for array

### ISSUE THREAD

**Issue by m-toman** [CSTR-Edinburgh/magphase#6]
Error:

> 3032 Traceback (most recent call last):
> 3033   File "0_batch_feature_extraction_for_merlin.py", line 63, in <module>
> 3034     feat_extraction(in_wav_dir, token, out_feats_dir)
> 3035   File "0_batch_feature_extraction_for_merlin.py", line 40, in feat_extraction
> 3036     mp.analysis_compressed(wav_file, out_dir=out_feats_dir)
> 3037   File "/home/mtoman/MagPhase/src/magphase.py", line 1607, in analysis_compressed
> 3038     m_mag, m_real, m_imag, v_f0, fs, v_shift = analysis_lossless(wav_file, fft_len=fft_len)
> 3039   File "/home/mtoman/MagPhase/src/magphase.py", line 1582, in analysis_lossless
> 3040     v_pm_sec, v_voi = la.read_reaper_est_file(est_file, check_len_smpls=len(v_sig), fs=fs)
> 3041   File "/home/mtoman/MagPhase/src/libaudio.py", line 455, in read_reaper_est_file
> 3042     v_pm_sec  = m_data[:,0]
> 3043 IndexError: too many indices for array

In:
https://github.com/CSTR-Edinburgh/magphase/blob/master/src/libaudio.py#L452

Caused by an EST file
[temp_15857.est.txt](https://github.com/CSTR-Edinburgh/magphase/files/1478555/temp_15857.est.txt)
(the input sound sample looks pretty normal)

Potential solution:
Adding `ndmin=2` to `loadtxt` ensures same dimensions for all return values. 
But follows up with another error later on.
So perhaps we can return an error here and ignore the file in question?

**Comment by felipeespic** [comment]
It is strange, could you send me the wavefile, please?

**Comment by felipeespic** [comment]
I just ran the loadtxt function with your est file, and it worked well. I think that it is a problem related with the numpy version. Upgrade numpy to the newest version and see if it fixed.

**Comment by m-toman** [comment]
Thanks for the answer, the issue was/is the access to m_data[:, 0], because numpy reads the file as a vector:
```
>>> import numpy as np
>>> np.version.version
'1.12.1'
>>> est_file = "temp_15857.est.txt"
>>> skiprows=7 
>>> usecols=[0,1]
>>> m_data = np.loadtxt(est_file, skiprows=skiprows, usecols=usecols)
>>> m_data.shape
(2,)
>>> v_pm_sec  = m_data[:,0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: too many indices for array
```

Updated numpy but still the same:
```
>>> np.version.version
'1.13.3'
>>> m_data = np.loadtxt(est_file, skiprows=skiprows, usecols=usecols)
>>> m_data.shape
(2,)
>>> v_pm_sec  = m_data[:,0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: too many indices for array
```

As mentioned above, adding ndmin=2 to the call forces a matrix, but I got an exception a bit later. Anyways, there is something wrong with the EST file only containing 1 line anyway.
I'll see if I can find a wav where that happens (I've just added a try/catch to the extraction script for now)

**Comment by felipeespic** [comment]
I got it. Fixed in a966befabbf116b3f1d079c48f06e24ec65984d1
Thanks
